src/main/java/org/jspecify/annotations/Nullable.java

```diff
@@ -42,88 +42,95 @@
- *       extends @Nullable Object>}. This means that a <i>type argument</i> supplied for that type
- *       parameter is permitted to be nullable if desired: {@code List<@Nullable String>}. (A
- *       non-null type argument, as in {@code List<String>}, is permitted either way.)
- *   <li>On a usage of a <b>type variable</b>: A type parameter, like the {@code E} in {@code
- *       interface List<E>}, defines a "type variable" of the same name, usable only <i>within</i>
- *       the scope of the declaring API element. In any example using {@code String} above, a type
- *       variable like {@code E} might appear instead. {@code @Nullable} continues to mean "or null"
- *       as always, but notably, this works without regard to whether the type argument is
- *       <i>already</i> nullable. For example, suppose that {@code class Foo<E extends @Nullable
- *       Object>} has a method {@code @Nullable E eOrNull()}. Then, whether {@code foo} is of type
+ *       extends @Nullable Object>}. This means that a type argument supplied for that type
+ *       parameter can <i>optionally</i> be nullable: {@code List<@Nullable String>}. (A non-null
+ *       type argument, as in {@code List<String>}, is permitted either way.)
+ *   <li>On a usage of a <b>type variable</b>: A type parameter, like the {@code E} declared in
+ *       {@code interface List<E>}, introduces a <i>type variable</i> of the same name, usable only
+ *       within the scope of the declaring API element. All the preceding examples using {@code
+ *       String} are equally valid for a type variable like {@code E}. {@code @Nullable} continues
+ *       to mean "or null" as always, regardless of whether the type variable "already" includes
+ *       {@code null}. For example, suppose that {@code class Foo<E extends @Nullable Object>} has a
+ *       method {@code @Nullable E eOrNull()}. Then, whether a variable {@code foo} is of type
  *       {@code Foo<String>} or {@code Foo<@Nullable String>}, the expression {@code foo.eOrNull()}
- *       is nullable either way. Using {@code @Nullable E} in this way is called "nullable
- *       projection" (<a href="NonNull.html#projection">non-null projection</a> is likewise
- *       supported, but less commonly useful).
+ *       is nullable either way. Using {@code @Nullable E} in this way is called <i>nullable
+ *       projection</i> (the more rarely needed <a href="NonNull.html#projection">non-null
+ *       projection</a> is also supported).
+ *   <li>On an <b>array type</b>: The nullness of the array and of its components can both be
+ *       indicated separately. For example, in null-marked code, {@code @Nullable String[]} is a
+ *       non-null array of nullable strings, whereas {@code String @Nullable []} is a nullable array
+ *       of non-null strings. Note that arrays are treated as covariant: {@code String[]} (array of
+ *       non-null strings) is assignable to {@code @Nullable String[]} (array of nullable strings).
```

Comment:
This appears to be the only place in the Javadoc where we mention subtyping at all, let alone covariance. Does it belong here?

Reply:
My first reaction is that this suggests we should be talking about it more.

Reply:
We should, but in the Javadoc?